pydicom/dataset.py

```diff
@@ -2030,19 +2095,21 @@ def __str__(self):
         """
         return self._pretty_str()
 
-    def top(self):
+    def top(self) -> str:
         """Return a :class:`str` representation of the top level elements. """
         return self._pretty_str(top_level_only=True)
 
-    def trait_names(self):
+    def trait_names(self) -> List[str]:
         """Return a :class:`list` of valid names for auto-completion code.
 
         Used in IPython, so that data element names can be found and offered
         for autocompletion on the IPython command line.
         """
-        return dir(self)  # only valid python >=2.6, else use self.__dir__()
+        return dir(self)
 
-    def update(self, dictionary):
+    def update(
```

Comment:
Note to self: does this work properly if the dictionary keys are element keywords?